torch/csrc/autograd/input_buffer.cpp

```diff
@@ -118,7 +118,12 @@ static void accumulate(
       can_accumulate_inplace(old_var) && !at::isTensorSubclassLike(var)) {
     buffer[pos] = old_var.add_(var);
   } else {
-    buffer[pos] = old_var + var;
+    // NestedTensors don't support addition between tensors with different striding patterns.
+    if (old_var.is_nested()){
+      buffer[pos] = old_var.contiguous() + var.contiguous();
```

Comment:
Let's add this logic to `add` directly so that we can special case for shapes and conditions. Then we can also throw a warning when a user encounters this.